.github/scripts/wg-onboard.sh

```
@@ -0,0 +1,750 @@
+  ssh_cmd bash -c "test -f $(remote_quote "$conf")" 2>/dev/null
+}
+
+# ---- Flock-guarded peer allocation/record on the jumpserver ----------------
+# Selection and assigned.txt recording happen under one lock to prevent races.
+atomic_allocate_peers() {
+  # SSH omits empty argv entries; placeholders keep arg positions stable for bash -s.
+  local force_tf="${TF_PEER:-__none__}"
+  local force_wg0="${WG0_PEER:-__none__}"
+  local force_wg1="${WG1_PEER:-__none__}"
+  ssh_cmd bash -s -- \
+    $(remote_quote "$ASSIGNED_FILE") \
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)